src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -728,6 +728,691 @@ public object WProp
+                );
+        }
+
+        [Fact]
+        public void TestWellKnownAttributeOnProperty_FieldOffset3()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+
+[StructLayout(LayoutKind.Auto)]
+public class Test
+{
+    [field: FieldOffset(4)]
+    public int P { get; set; }
+}
+";
+            var comp = CreateStandardCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,13): error CS0636: The FieldOffset attribute can only be placed on members of types marked with the StructLayout(LayoutKind.Explicit)
+                //     [field: FieldOffset(4)]
+                Diagnostic(ErrorCode.ERR_StructOffsetOnBadStruct, "FieldOffset").WithLocation(7, 13)
+                );
+        }
+
+        [Fact]
+        public void TestWellKnownAttributeOnProperty_FieldOffset4()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+
+public struct Test
+{
+    [field: FieldOffset(4)]
+    public int P { get; set; }
+}
+";
+            var comp = CreateStandardCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,13): error CS0636: The FieldOffset attribute can only be placed on members of types marked with the StructLayout(LayoutKind.Explicit)
+                //     [field: FieldOffset(4)]
+                Diagnostic(ErrorCode.ERR_StructOffsetOnBadStruct, "FieldOffset").WithLocation(6, 13)
+                );
+        }
+
+        [Fact]
+        public void TestWellKnownAttributeOnProperty_FieldOffset5()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+
+[StructLayout(LayoutKind.Explicit)]
+public struct Test
+{
+    public int P { get; set; }
+}
+";
+            var comp = CreateStandardCompilation(source);
+            comp.VerifyDiagnostics(
+                // (7,16): error CS0625: 'Test.<P>k__BackingField': instance field types marked with StructLayout(LayoutKind.Explicit) must have a FieldOffset attribute
+                //     public int P { get; set; }
+                Diagnostic(ErrorCode.ERR_MissingStructOffset, "P").WithArguments("Test.<P>k__BackingField").WithLocation(7, 16)
+                );
+        }
+
+        [Fact]
+        public void TestWellKnownAttributeOnProperty_FixedBuffer()
+        {
+            string source = @"
+public class Test
+{
+    [field: System.Runtime.CompilerServices.FixedBuffer(typeof(int), 0)]
+    public int P { get; set; }
+}
+";
+            var comp = CreateStandardCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,13): error CS1716: Do not use 'System.Runtime.CompilerServices.FixedBuffer' attribute. Use the 'fixed' field modifier instead.
+                //     [field: System.Runtime.CompilerServices.FixedBuffer(typeof(int), 0)]
+                Diagnostic(ErrorCode.ERR_DoNotUseFixedBufferAttr, "System.Runtime.CompilerServices.FixedBuffer").WithLocation(4, 13)
```

Comment:
>Diagnostic(ErrorCode.ERR_DoNotUseFixedBufferAttr, "System.Runtime.CompilerServices.FixedBuffer").WithLocation(4, 13) [](start = 16, length = 116)

Should we report a slightly different error? A fixed modifier cannot be applied to a property.